scrapyjs/middleware.py

```diff
@@ -43,24 +52,26 @@ def from_crawler(cls, crawler):
         return cls(crawler, splash_base_url, slot_policy)
 
     def process_request(self, request, spider):
-        splash_options = request.meta.get('splash')
+        splash_options = self.get_splash_options(request, spider)
         if not splash_options:
             return
 
+        elif request.meta.get("_splash_processed"):
+            return
+
         if request.method != 'GET':
             log.msg("Currently only GET requests are supported by SplashMiddleware; %s "
                     "will be handled without Splash" % request, logging.WARNING)
             return request
 
         meta = request.meta
-        del meta['splash']
-        meta['_splash_processed'] = splash_options
```

Comment:
This is related to https://github.com/scrapinghub/scrapyjs/pull/15/files#r28138323 (avoiding generating multiple splash requests). If splash options were in request meta as content of `splash` key they can be deleted and when request will be processed by middleware second time process_request will return None. But if they are set per spider you can't easily delete them. This means that first check on line 55 will always return `True` if options are in spider. This means we need another check, so I needed to redefine `_splash_processed` key, changing it's content from dictionary of splash options to Boolean flag.